app/workers/scheduler/user_registration_protection_scheduler.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class Scheduler::UserRegistrationProtectionScheduler
+  include Sidekiq::Worker
+
+  def perform
+    return unless User.where(created_at: (Time.now.utc - 5.minutes)..Time.now.utc).count > ENV.fetch('MAX_REGISTRATIONS_THRESHOLD', 1_000).to_i
```

Comment:
I don't think the query needs to be a range as it's just needing the count of all accounts since 5 minutes ago, which would have a natural upper bounds at the current time.
 
```suggestion
    if User.where('created_at >= :start', {
      start: Time.now.utc - max_registrations_window
    }).count > max_registrations_threshold
```

And then moving the fetching of the environment variable into separate methods (as to differentiate between the configuration and functionality). I've also introduced a `max_registrations_window` variable there, as to allow a server admin to configure this for different timeframes

Additionally, I'm thinking when the registration mode is changed like this, the administrators & moderators should be emailed / the audit log should be updated with that change having taken place, such that they're aware that the server's registration mode has changed.